composer/trainer/trainer.py

```diff
@@ -1537,6 +1528,12 @@ def _train_loop(self) -> None:
                     # `now` is actually in the past, but want to include the time it takes to perform this reduction
                     last_wct = now
 
+                    if self.train_metrics is not None:
+                        self._compute_and_log_metrics(
+                            dataloader_label='train',
+                            metrics=self.train_metrics,
+                        )
+
```

Comment:
Why is this moved? Is it because you want the older timestamp?

Reply:
yes exactly. train_metrics had an off by one error